src/cmake/GeosxConfig.cmake

```diff
@@ -57,30 +57,3 @@ configure_file( ${CMAKE_SOURCE_DIR}/coreComponents/common/GeosxConfig.hpp.in
 
 install( FILES ${CMAKE_BINARY_DIR}/include/common/GeosxConfig.hpp
          DESTINATION ${CMAKE_INSTALL_PREFIX}/include/common )
-
-
-function( make_full_config_file
-          PREPROCESSOR_VARS )
-    foreach( DEP in ${PREPROCESSOR_VARS} )
-        set( USE_${DEP} TRUE )
-        set( GEOSX_USE_${DEP} TRUE )
-        set( ${DEP} TRUE )
-    endforeach()
-
-    # Fix some options to avoid changes in committed config for doxygen
-    set( GEOSX_CMAKE_BUILD_TYPE "\"Release\"" )
-    set( GEOSX_LOCALINDEX_TYPE "std::ptrdiff_t" )
-    set( GEOSX_LOCALINDEX_TYPE_FLAG "3" )
-    set( GEOSX_GLOBALINDEX_TYPE "long long int" )
-    set( GEOSX_GLOBALINDEX_TYPE_FLAG "2" )
-    set( GEOSX_LA_INTERFACE "Hypre" )
-    set( GEOSX_LA_INTERFACE_HYPRE ON )
-    set( GEOSX_LA_INTERFACE_TRILINOS OFF )
-    set( GEOSX_LA_INTERFACE_PETSC OFF )
-
-    configure_file( ${CMAKE_SOURCE_DIR}/coreComponents/common/GeosxConfig.hpp.in
-                    ${CMAKE_SOURCE_DIR}/docs/doxygen/GeosxConfig.hpp )
```

Comment:
The status after this patch would be that we'll end up with the out of date information.
This may not be what we want either.

I wonder if the simplest answer to this hassle is to generate `GeosxConfig.hpp` from `GeosxConfig.hpp.in` (like we're doing), but in the `generatedSrc` and not in the sources (I think) folder.
So it's not committed but up-to-date for all the builds.

That looks to me like a sensible thing to do.

Reply:
The goal of generating this file is for it to be accessible in `readthedocs` environment where doxygen docs are built and where we don't have a full-fledged build environment to run `cmake` in. It has to be committed to the repo, that's the whole point. Otherwise we are already generating one under `build/include/common/GeosxConfig.hpp`, it would do the job perfectly if only we could do it on `readthedocs`.

Yes, it will go out of date, but it only affects what the user sees when they browse doxygen online, it's not a big deal. Once in a while I will push an updated version, but even if I don't, not much harm.

Reply:
This file could also be an artefact from the ci where the cmake environment is set. Either used by a `documentation` target either another pipeline. See draft idea #2603

Reply:
> If we need to commit/push once in a while, what's the benefit w.r.t. directly editing [readthedocs](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/buildGuide/Dependencies.html) (which can even be done in the web ui, no need to tweak anything in your own computer)?

Would actually be nice if that page was built from the config file, but it's not (and it's severely out of date at this point :laughing:). `GeosxConfig.hpp` only shows up [here](https://geosx-geosx.readthedocs-hosted.com/en/latest/doxygen_output/html/_geosx_config_8hpp.html), but its presence makes all `#include`s of it in the source clickable and lets users read the "documentation" of each macro. It's just a fairly small convenience.